### PR TITLE
KFNBC: Admin Spawner Tab

### LINKED ISSUE #273: [KFNBC] Add admin section from jupyterhub spawner
Admins must be able to access the same functionality currently available via JH admin console (eg. view NB server sessions, stop sessions, etc.).

### PR DESCRIPTION
**Pull request by andrewballantyne**
Resolves #273 

TODOs:
- [x] Test and debug some desync of data issues; starting admin notebook, then stopping it, and then viewing the notebook server page - it shows the old information for a few moments before the data kicks and it refreshes
- [x] Admin table columns need to be sortable
- [x] Privilege column needs to be added
- [x] Add pagination to the admin user list
- [x] Can't impersonate yourself

### CODE REVIEW

File: frontend/src/pages/notebookController/screens/server/SpawnerPage.tsx [hunk 1]
```diff
@@ -373,6 +372,17 @@ const SpawnerPage: React.FC = React.memo(() => {
       loadError={loadError}
       empty={!images || images.length === 0}
     >
+      {impersonatingUser && (
+        <Alert
+          variant="info"
+          title={`You are impersonating user "${currentUserState.user}"`}
```
Comment (andrewballantyne) [inline frontend/src/pages/notebookController/screens/server/SpawnerPage.tsx:378]: Probably not the best wording -- need to look into this.

**Comment by samuelvl** [comment]
@andrewballantyne Can you resolve conflicts? I will review it after

**Comment by andrewballantyne** [comment]
Done -- there was minor non-invasive conflicts. Please review when you get a chance @samuelvl

**Review by samuelvl — CHANGES_REQUESTED** [review]
I can see the `Administrative` tab only for admin users (+1). However I have created a couple of notebooks for regular users:

```
$ oc get notebooks
NAME                          AGE
jupyter-nb-openldap-2duser1   7m20s
jupyter-nb-openldap-2duser2   4m4s

$ oc get pods                                            
NAME                                               READY   STATUS    RESTARTS   AGE
jupyter-nb-openldap-2duser1-0                      2/2     Running   0          7m22s
jupyter-nb-openldap-2duser2-0                      2/2     Running   0          4m5s
```

The notebooks are running fine, however I'm seeing this weird behavior:

image

User names are empty, if I check the dashboard CR I get this information under the status:

```
status:
  notebookControllerState:
  - lastActivity: 1659542088594
    lastSelectedImage: ""
    lastSelectedSize: ""
    user: ""
  - lastActivity: 1659542180482
    lastSelectedImage: ""
    lastSelectedSize: ""
    user: ""
  - lastActivity: 1659542395488
    lastSelectedImage: ""
    lastSelectedSize: ""
    user: ""
  - lastActivity: 1659542415494
    lastSelectedImage: ""
    lastSelectedSize: ""
    user: ""
  - lastActivity: 1659542639507
    lastSelectedImage: ""
    lastSelectedSize: ""
    user: ""
```

It' even more strange because every 3 minutes, 2 new empty users are added to this list (compare both screenshots):

image

I'm using this dashboard image built from PR code: `quay.io/samuvl/odh-dashboard:v2.0-64-g73cccda`

**Comment by andrewballantyne** [comment]
@samuelvl Hmmm... usernames being empty is not my doing 🤔 Not sure how you actually got those. We had a bug in our infrastructure where if you were fast enough to click the create button you could create a user mapping without it matching one existing -- thus giving empty names. I also put a check to block empty names before it updates the config. I'm a bit confused on why you're getting it. I'll retest and see if this somehow was a bi-product of a rebase that I lost some functionality (although I did not see any issue last night)

As for the admin tab for only one user -- unfortunately it'll lead to worse UX if I hide the tab when it is only one user -- as the code will pull for more each time and at a random time (from the user point of view) it will snap in. I think we should keep it for now. We can discuss UX options with Kyle if needed.

**Comment by andrewballantyne** [comment]
It should be noted, we do not search for notebooks on the Admin page -- at least not as a primary thing. We search for those user data in the ODH Dashboard Config and compare it with those that have Notebooks. I suppose I could look at union-ing them... but I'm not sure 🤔 The notebooks has a translated name and will make it hard to undo that. We'd have to make an effort at a different level to do that.

**Review by DaoDaoNoCode — APPROVED** [review]
lgtm

**Comment by dlabaj** [comment]
Seeing the same thing is Samu. If you sort it more users get added.

**Comment by andrewballantyne** [comment]
I pushed a commit to avoid the problem @samuelvl and @dlabaj are having with the "users growing" and showing the empty user names (that is bad state generated from a previous version of the Dashboard -- fixed in my PR)

**Review by dlabaj — APPROVED** [review]

**Pull request merged**